react/src/components/atoms/forms/InputNumber/index.js

```diff
@@ -19,9 +19,18 @@ const NumberInput = (props) => (
     <InputContext.Consumer>
       {
         (context) => {
+          const hasValue = context.value && context.value !== 'NaN';
```

Comment:
You can't do what's currently there because context.value can be a number.
```suggestion
              let stringValue;
              if (typeof context.value !== 'string') {
                stringValue = String(context.value);
              } else {
                stringValue = context.value;
              }
              const numberValue = Number(stringValue);
              const hasValue = !Number.isNaN(numberValue) && stringValue.length > 0);
```